projects/openapi-cli/src/commands/verify.ts

```diff
@@ -0,0 +1,301 @@
+    .action(async (specPath) => {
+      const analytics: { event: string; properties: any }[] = [];
+
+      const absoluteSpecPath = Path.resolve(specPath);
+      if (!(await fs.pathExists(absoluteSpecPath))) {
+        return await feedback.inputError(
+          'OpenAPI specification file could not be found',
+          InputErrors.SPEC_FILE_NOT_FOUND
+        );
+      }
+
+      console.log('');
+      const options = command.opts();
+
+      const makeInteractionsIterator = async () =>
+        getInteractions(options, specPath, feedback);
+
+      /// Add if --document or --update options passed
+      if (options.document || options.patch) {
+        if (options.document) {
+          const specReadResult = await readDeferencedSpec(absoluteSpecPath);
+          if (specReadResult.err) {
+            await feedback.inputError(
+              `OpenAPI specification could not be fully resolved: ${specReadResult.val.message}`,
+              InputErrors.SPEC_FILE_NOT_READABLE
+            );
+          }
+          const { jsonLike: spec, sourcemap } = specReadResult.unwrap();
+
+          feedback.notable('Documenting operations...');
+
+          let { observations } = matchInteractions(
+            spec,
+            await makeInteractionsIterator()
+          );
+
+          const result = await addIfUndocumented(
+            options.document,
+            observations,
+            await makeInteractionsIterator(),
+            spec,
+            sourcemap
+          );
+
+          if (result.ok) {
+            analytics.push({
+              event: 'openapi.verify.document',
+              properties: {
+                allFlag: options.document === 'all',
+                numberDocumented: result.val.length,
+              },
+            });
+            result.val.map((operation) => {
+              console.log(
+                `${chalk.green('added')}  ${operation.method} ${
+                  operation.pathPattern
+                }`
+              );
+            });
+          }
+        }
+
+        if (options.patch) {
+          feedback.notable('Patching operations...');
+          const specReadResult = await readDeferencedSpec(absoluteSpecPath);
+          if (specReadResult.err) {
+            await feedback.inputError(
+              `OpenAPI specification could not be fully resolved: ${specReadResult.val.message}`,
+              InputErrors.SPEC_FILE_NOT_READABLE
+            );
+          }
+          const { jsonLike: spec, sourcemap } = specReadResult.unwrap();
+          const patchInteractions = await makeInteractionsIterator();
+          const patchStats = await patchOperationsAsNeeded(
+            patchInteractions,
+            spec,
+            sourcemap
+          );
+          analytics.push({
+            event: 'openapi.verify.patch',
+            properties: patchStats,
+          });
+        }
+
+        console.log(chalk.gray('-'.repeat(process.stdout.columns) + '\n'));
```

Comment:
What's the difference between `feedback` and directly logging to the console? I would have the feeling that sending everything to `feedback` to determine the level of logging etc would make sense, but I'm not sure I have the big picture in mind.

Reply:
There shouldn't be a difference. I'll pass the feedback around and centralize these calls